rust/connlib/tunnel/src/peer.rs

```diff
@@ -1464,24 +1464,34 @@ mod proptests {
         filters()
             .prop_filter("empty filters accepts every packet", |f| !f.is_empty())
             .prop_flat_map(|f| {
-                let filters = f.clone();
                 any::<ProtocolKind>()
-                    .prop_filter_map(
-                        "If ICMP is contained there is no way to generate gaps",
+                    .prop_filter_map("If ICMP is contained there is no way to generate gaps", {
+                        let f = f.clone();
+
                         move |p| {
-                            (p != ProtocolKind::Icmp || !filters.contains(&Filter::Icmp))
-                                .then_some(p)
-                        },
-                    )
+                            (p != ProtocolKind::Icmp || !f.contains(&Filter::Icmp)).then_some(p)
+                        }
+                    })
+                    .prop_filter("no gaps in port ranges", {
+                        let f = f.clone();
+
+                        move |p| !gaps(f.clone(), *p).is_empty()
+                    })
```

Comment:
This is the important part. Without this, we would sometimes have port ranges that don't have gaps and then we fail further down when we continue to sample.